pkg/sql/schemachanger/scplan/internal/rules/dep_drop_object.go

```diff
@@ -26,8 +26,36 @@ import (
 //   reaches the ABSENT state, but afterwards in the stage, so as to not
 //   interfere with the event logging op which is tied to the descriptor element
 //   removal.
+
 func init() {
 
+	registerDepRule(
+		"descriptor txn_drop before descriptor drop",
+		scgraph.PreviousStagePrecedence,
+		"descriptor", "dependent",
```

Comment:
If this rule is about making sure one descriptor reached `TXN_DROPPED` before reaching `DROPPED` in a previous stage, then we probably should change the name of the second variable `dependent` to, maybe, `descriptor_self`.

Reply:
Good catch, fixed